### PR TITLE
Update Dev Guide for stats command and ui 

### PR DESCRIPTION
**Pull request by limli**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -447,6 +447,73 @@ In other words, when RestaurantBook#deleteMember() or RestaurantBook#deleteIngre
 
 * Decision: Alternative 1 was adopted as it is an easier implementation and it makes it more convenient for the user to delete a record. It is intuitive that if a deleteMember or deleteIngredient command is called and these objects cease to exist in the restaurant book, the corresponding Booking or Recipe containing these objects should also cease to exist. To deal with the possibility that the member or ingredient is accidentally deleted, and all referenced bookings and recipes is unintentionally deleted, an additional field in the delete command can be added as a confirmation that these records should be deleted.
 
+=== The `stats` command
+
+When the `stats` command is entered, the current statictis of the amount of bookings is to be displayed to the user in the form of a bar chart.
```
Comment (lingyanhao) [inline docs/DeveloperGuide.adoc]: "statistics"- fix typo
"amount of bookings"- what does this even mean

@limli bear in mind you'll have to explain this in the user guide too

if you're unsure what you want right now, probably ok. but remember to give a clear specification of what the command does.

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -447,6 +447,73 @@ In other words, when RestaurantBook#deleteMember() or RestaurantBook#deleteIngre
...
+
+==== Current implementation
+
+The `RestaurantBookParser` class would detect if the user has keyed in the keyword `stats`. If it does, `RestaurantBookParser` would call the `Statistics` class. The `Statistics` class is used to determine what data is to be displayed to the user in the bar chart (ie what are the axes, values, and legends of the bar chart). It would call the `StatsWindow` class which handles the UI. The bar chart will appear in a new window and does not affect the current window.
+
```
Comment (lingyanhao) [inline docs/DeveloperGuide.adoc:457]: Would be a good place for sequence diagram.

Also, unnecessarily long wording.
The `RestaurantBookParser` class would detect if the user has keyed in the keyword `stats`. If it does, `RestaurantBookParser` would call the `Statistics` class.

-> When the user enters the `stats` command, `RestaurantBookParser` calls the `Statistics` class.

The `Statistics` class is used to determine what data is to be displayed to the user in the bar chart

-> The `Statistics` class fetches the relevant data for user display

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -447,6 +447,73 @@ In other words, when RestaurantBook#deleteMember() or RestaurantBook#deleteIngre
...
+* Decision: Alternative 1 was adopted as it is an easier implementation. As the number of times the `stats` command is called is likely to be low as compared to an `add` or `delete` operation, doing the calculation after every operation would be a waste of computation power. 
+
+===== Aspect: Design of UI display
+
+* **Alternative 1 (current choice):** Displaying the statistics in a new window.
+** Pros: The statistics would be displayed independently from the database and would not interfere with each other. 
+** Cons: The user can open multiple windows which contains different data and not know which is the correct window.
+
+* **Alternative 2:** Hide the displayed database to show the statistics.
```
Comment (lingyanhao) [inline docs/DeveloperGuide.adoc]: With a separate command to toggle back and forth?

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -447,6 +447,73 @@ In other words, when RestaurantBook#deleteMember() or RestaurantBook#deleteIngre
...
+** Pros: There would be only 1 window to manage at any point in time.
+** Cons: Cannot view the statistics and the database concurrently.
+
+* Decision: Alternative 1 was adopted as it is an easier implementation. Implementing the `stats` window as a static window that will not change would not affect the other parts of the app, and there is no need to hide and show the current screen. Moreover, the user might want to view the statistics together with the database, which is only possible with multiple windows.
```
Comment (lingyanhao) [inline docs/DeveloperGuide.adoc:482]: Easier to implement- not mentioned in the pros and cons analysis? Try not to suddenly bring in a new point like that- bad logical flow.

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -447,6 +447,73 @@ In other words, when RestaurantBook#deleteMember() or RestaurantBook#deleteIngre
...
+
+===== Aspect: Implementation of data structure
+
+* **Alternative 1 (current choice):** Doing the calculation only when the user wants to view the statistics.
+** Pros: Calculating only when needed is more efficient.
+** Cons: The command would take some time to calculate before being able to display the statistics.
+
+* **Alternative 2:** Every change to the database updates the statistics on the fly.
+** Pros: Displaying the statistics on the UI would be faster.
+** Cons: Need to change the implementation of every command that causes a change to the database. Calculations have to be done every operation, affecting efficiency.
```
Comment (lingyanhao) [inline docs/DeveloperGuide.adoc]: Conflicting ideas of efficiency? So which is more efficient? Potentially confusing.

File: docs/DeveloperGuide.adoc [hunk 6]
```diff
@@ -447,6 +447,73 @@ In other words, when RestaurantBook#deleteMember() or RestaurantBook#deleteIngre
...
+
+=== Current implementation
+
+The panel is formatted in the `ItemListPanel.fxml` file, and each item has its own controller (`MemberListPanel`, `StaffListPanel`, `BookingListPanel`, and `IngredientListPanel`) that extends from `ItemListPanel`. The panel displays the database via cards. Each item has its own card (`MemberCard`, `StaffCard`, `BookingCard`, and `IngredientCard`).
+
+==== Design Considerations
+
+===== Aspect: Design of card display
+
+* **Alternative 1 (current choice):** Making a separate fxml file for each item card.
```
Comment (lingyanhao) [inline docs/DeveloperGuide.adoc]: For each item card vs for each type of item card

If I have 100 members in the restaurant-> 100 fxml files?

File: docs/DeveloperGuide.adoc [hunk 7]
```diff
@@ -447,6 +447,73 @@ In other words, when RestaurantBook#deleteMember() or RestaurantBook#deleteIngre
...
+
+* **Alternative 1 (current choice):** Using the same fxml file for each item panel.
+** Pros: Less duplication.
+** Cons: Hard to customise the contents of each panel. 
+
+* **Alternative 2:** Making a separate fxml file for each item panel
+** Pros: Able to customise the panel to suit the item.
+** Cons: Need to duplicate common elements that all the item panels share.
+
+* Decision: Alternative 1 was adopted. Currently, the only difference between the panels are the header which contains the name of the item type. This only affects a single label, which is easy to edit via code. Hence, making them share the fxml reduces the code duplication. If the display of the panels were to be modified in the future, say to add more padding or to change the size, only one fxml file would need to be modified.
```
Comment (lingyanhao) [inline docs/DeveloperGuide.adoc]: So is the header a label?

making them share the fxml -> sharing the fxml
You really need to kill unnecessary words

**Comment by hiyingnn** [comment]
Currently, the headings for current implementation under panel UI for Member, Staff, Booking, and Ingredient is incorrect.
3.7. The panel UI for Member, Staff, Booking, and Ingredient
3.8. Current implementation --> should be 3.7.1 and should not appear on the directory

**Pull request merged**